assignments/array-methods.js

```diff
@@ -54,30 +57,89 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first 
+// and last names into a new array called fullName. 
 let fullName = [];
+
+runners.forEach(function (runner) {
+    fullName.push(`${runner.first_name} ${runner.last_name}`);
+});
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+// The event director needs to have all the runner's first names converted to uppercase because the director 
+// BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
+// let allCaps = [];
+
+const bigFirstNames = runners.map(function(runner){
+    return runner.first_name.toUpperCase();
+});
+
+console.log(bigFirstNames); 
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with 
+// large sized shirts so they can choose a different size. Return an array named largeShirts that contains information 
+// about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+
+// pretty sure this is not the correct usage
+runners.filter(function(runner) {
+    if (runner.shirt_size === 'L') {
+      largeShirts.push(runner); 
+    }
+  });
```

Comment:
compare this to the code solution:
let largeShirts = runners.filter((shirts) => {
  return shirts.shirt_size === "L";
});
console.log(largeShirts);

notice we don't need an "if" statement in filter